NPE on getFolders method in PortletWebViewActivity

E/AndroidRuntime(21726): java.lang.RuntimeException: Unable to start activity ComponentInfo{org.apereo/org.apereo.activities.PortletWebViewActivity_}: java.lang.NullPointerException
E/AndroidRuntime(21726):        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2544)
E/AndroidRuntime(21726):        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2603)
E/AndroidRuntime(21726):        at android.app.ActivityThread.access$900(ActivityThread.java:174)
E/AndroidRuntime(21726):        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1321)
E/AndroidRuntime(21726):        at android.os.Handler.dispatchMessage(Handler.java:102)
E/AndroidRuntime(21726):        at android.os.Looper.loop(Looper.java:146)
E/AndroidRuntime(21726):        at android.app.ActivityThread.main(ActivityThread.java:5752)
E/AndroidRuntime(21726):        at java.lang.reflect.Method.invokeNative(Native Method)
E/AndroidRuntime(21726):        at java.lang.reflect.Method.invoke(Method.java:515)
E/AndroidRuntime(21726):        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1291)
E/AndroidRuntime(21726):        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1107)
E/AndroidRuntime(21726):        at dalvik.system.NativeStart.main(Native Method)
E/AndroidRuntime(21726): Caused by: java.lang.NullPointerException
E/AndroidRuntime(21726):        at org.apereo.activities.PortletWebViewActivity.initialize(PortletWebViewActivity.java:112)
E/AndroidRuntime(21726):        at org.apereo.activities.PortletWebViewActivity_.onViewChanged(PortletWebViewActivity_.java:94)
E/AndroidRuntime(21726):        at org.androidannotations.api.view.OnViewChangedNotifier.notifyViewChanged(OnViewChangedNotifier.java:41)
E/AndroidRuntime(21726):        at org.apereo.activities.PortletWebViewActivity_.setContentView(PortletWebViewActivity_.java:58)
E/AndroidRuntime(21726):        at org.apereo.activities.PortletWebViewActivity_.onCreate(PortletWebViewActivity_.java:46)
E/AndroidRuntime(21726):        at android.app.Activity.performCreate(Activity.java:5600)
E/AndroidRuntime(21726):        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1093)
E/AndroidRuntime(21726):        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2508)
E/AndroidRuntime(21726):        ... 11 more

--- a/app/src/main/java/org/apereo/activities/HomePageActivity.java
+++ b/app/src/main/java/org/apereo/activities/HomePageActivity.java
@@ -93,9 +93,14 @@ public class HomePageActivity extends BaseActivity implements IActionListener, A
         // set a custom shadow that overlays the main content when the drawer opens
         mDrawerLayout.setDrawerShadow(R.drawable.drawer_shadow, GravityCompat.START);
         // set up the drawer's list view with items and click listener
-        List<Folder> folders = layoutManager.getLayout().getFolders();
-        mDrawerList.setAdapter(new FolderListAdapter(this,
-                R.layout.drawer_list_item, folders, ePosition));
+        try {
+            List<Folder> folders = layoutManager.getLayout().getFolders();
+            mDrawerList.setAdapter(new FolderListAdapter(this,
+                    R.layout.drawer_list_item, folders, ePosition));
+        } catch (NullPointerException e) {
+            Logger.d(TAG, e.getMessage());
+            LaunchActivity_.intent(this);
+        }
         mDrawerList.setOnItemClickListener(this);
 
         // ActionBarDrawerToggle ties together the the proper interactions
@@ -222,7 +227,14 @@ public class HomePageActivity extends BaseActivity implements IActionListener, A
 
         // update selected item and title, then close the drawer
         mDrawerList.setItemChecked(position, true);
-        setTitle(layoutManager.getLayout().getFolders().get(position).getName());
+
+        // Workaround for layoutManager being possibly garbage collected
+        try {
+            setTitle(layoutManager.getLayout().getFolders().get(position).getName());
+        } catch (NullPointerException e) {
+            Logger.d(TAG, e.getMessage());
+            LaunchActivity_.intent(this);
+        }
 
         mDrawerLayout.closeDrawer(mDrawerList);
     }
@@ -274,10 +286,16 @@ public class HomePageActivity extends BaseActivity implements IActionListener, A
         // Update the login/logout button.
         invalidateOptionsMenu();
 
-        // Update the side drawer contents.
-        List<Folder> folders = layoutManager.getLayout().getFolders();
-        mDrawerList.setAdapter(new FolderListAdapter(this,
-                R.layout.drawer_list_item, folders, ePosition));
+        // Workaround for layoutManager being possibly garbage collected
+        try {
+            // Update the side drawer contents.
+            List<Folder> folders = layoutManager.getLayout().getFolders();
+            mDrawerList.setAdapter(new FolderListAdapter(this,
+                    R.layout.drawer_list_item, folders, ePosition));
+        } catch (NullPointerException e) {
+            Logger.d(TAG, e.getMessage());
+            LaunchActivity_.intent(this);
+        }
 
         // Switch back to the first tab.
         selectItem(0);

--- a/app/src/main/java/org/apereo/activities/HomePageActivity.java
+++ b/app/src/main/java/org/apereo/activities/HomePageActivity.java
@@ -1,8 +1,6 @@
 package org.apereo.activities;
 
 import android.app.Fragment;
-import android.app.FragmentManager;
-import android.app.FragmentTransaction;
 import android.content.Intent;
 import android.content.res.Configuration;
 import android.os.Bundle;
@@ -222,8 +220,10 @@ public class HomePageActivity extends BaseActivity implements IActionListener, A
         Fragment fragment = HomePageListFragment.getFragment(this);
         fragment.setArguments(args);
 
-        FragmentManager fragmentManager = getFragmentManager();
-        fragmentManager.beginTransaction().replace(R.id.content_frame, fragment).commit();
+        getFragmentManager()
+                .beginTransaction()
+                .replace(R.id.content_frame, fragment)
+                .commit();
 
         // update selected item and title, then close the drawer
         mDrawerList.setItemChecked(position, true);
@@ -277,11 +277,11 @@ public class HomePageActivity extends BaseActivity implements IActionListener, A
         Fragment fragment = HomePageListFragment.getFragment(this);
         fragment.setArguments(args);
 
-        FragmentManager fm = getFragmentManager();
-        FragmentTransaction ft = fm.beginTransaction();
-        ft.setCustomAnimations(android.R.animator.fade_in, android.R.animator.fade_out);
-        ft.replace(R.id.content_frame, fragment);
-        ft.commit();
+        getFragmentManager()
+                .beginTransaction()
+                .setCustomAnimations(android.R.animator.fade_in, android.R.animator.fade_out)
+                .replace(R.id.content_frame, fragment)
+                .commit();
 
         // Update the login/logout button.
         invalidateOptionsMenu();

--- a/app/src/main/java/org/apereo/activities/PortletWebViewActivity.java
+++ b/app/src/main/java/org/apereo/activities/PortletWebViewActivity.java
@@ -32,6 +32,7 @@ import org.apereo.R;
 import org.apereo.adapters.FolderListAdapter;
 import org.apereo.models.Folder;
 import org.apereo.utils.LayoutManager;
+import org.apereo.utils.Logger;
 
 import java.util.List;
 
@@ -108,8 +109,17 @@ public class PortletWebViewActivity extends BaseActivity implements AdapterView.
     void initialize() {
         // set a custom shadow that overlays the main content when the drawer opens
         mDrawerLayout.setDrawerShadow(R.drawable.drawer_shadow, GravityCompat.START);
+
+        List<Folder> folders = null;
+        // workaround for layoutManager being possibly garbage collected
+        try {
+            folders = layoutManager.getLayout().getFolders();
+        } catch (NullPointerException e) {
+            Logger.d(TAG, e.getMessage());
+            LaunchActivity_.intent(this);
+        }
+
         // set up the drawer's list view with items and click listener
-        List<Folder> folders = layoutManager.getLayout().getFolders();
         mDrawerList.setAdapter(new FolderListAdapter(this,
                 R.layout.drawer_list_item, folders, folderPosition));
         mDrawerList.setOnItemClickListener(this);

--- a/app/src/main/java/org/apereo/services/RestApi.java
+++ b/app/src/main/java/org/apereo/services/RestApi.java
@@ -18,8 +18,10 @@ import org.apache.http.client.methods.HttpGet;
 import org.apache.http.impl.client.DefaultHttpClient;
 import org.apereo.App;
 import org.apereo.R;
+import org.apereo.activities.LaunchActivity_;
 import org.apereo.utils.Logger;
 import org.springframework.http.client.SimpleClientHttpRequestFactory;
+import org.springframework.web.client.ResourceAccessException;
 import org.springframework.web.client.RestTemplate;
 
 import java.io.BufferedReader;
@@ -85,6 +87,8 @@ public class RestApi {
             powerClient.getMainFeed();
             String response = getResponse(client, httpGet);
             callbackHandler.onSuccess(callback, response);
+        } catch (ResourceAccessException rae) {
+            LaunchActivity_.intent(App.getInstance());
         } catch (Exception e) {
             callbackHandler.onError(callback, e, "getMainFeed");
         } finally {

